### PR TITLE
[13.0] l10n_pe,l10n_latam_invoice_document: update

### PR DESCRIPTION
**Pull request by umiphos**
Description of the issue/feature this PR addresses:

First, l10n_pe does not need to have base_vat, we are adding the
dependency l10n_latam_base, there is the dependency of base_vat.

Second, base_vat does not need to check the structure and letters of the
VAT because we are already adding a field for the type of document in
l10n_latam_base(l10n_latam_identification_type_id), this field does what
the logic in the module did before.

Now, all the future developments should only use the VAT and
l10n_latam_identification_type_id and not split_vat, that method is no
longer useful is we use the latam modules.

Previous PR for  historic purposes https://github.com/Vauxoo/enterprise/pull/569

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

File: addons/l10n_pe/__manifest__.py [hunk 1]
```diff
@@ -11,6 +11,7 @@
         'base_address_extended',
         'base_address_city',
         'l10n_latam_base',
+        'l10n_latam_invoice_document',
```
Comment (jco-odoo) [inline addons/l10n_pe/__manifest__.py]: We will let depend l10n_pe_edi then.

File: addons/l10n_pe/models/res_company.py [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+
+
+class ResCompany(models.Model):
+
+    _inherit = "res.company"
+
+    @api.onchange('country_id')
+    def onchange_country(self):
+        """In Peru, the rounding method it's calculated as global"""
+        for rec in self.filtered(lambda x: x.country_id == self.env.ref('base.pe')):
```
Comment (jco-odoo) [inline addons/l10n_pe/models/res_company.py]: should be done on CoA install (same was done for cl/ar)

File: addons/l10n_pe/models/account_chart_template.py [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+
+    def _prepare_all_journals(self, acc_template_ref, company, journals_dict=None):
+        """ If Peruvian chart, we don't create sales journal as we need more data to create it properly """
+        journals = super()._prepare_all_journals(acc_template_ref, company, journals_dict=journals_dict)
+        if company.country_id == self.env.ref('base.pe'):
+            for journal in journals:
+                if journal['type'] == 'sale':
+                    journal.update({'l10n_latam_use_documents': True,
+                                    'default_debit_account_id': self.env.ref('l10n_pe.chart1213').id,
+                                    'default_credit_account_id': self.env.ref('l10n_pe.chart1213').id})
```
Comment (jco-odoo) [inline addons/l10n_pe/models/account_chart_template.py]: Move to edi module too somehow?

File: addons/base_vat/models/res_partner.py [hunk 4]
```diff
@@ -326,37 +326,30 @@ def check_vat_no(self, vat):
...
-                return False
-
-            for f in range(0, 10):
-                sum += int(factor[f]) * int(vat[f])
+        # verify RUC
+        factor = '5432765432'
+        sum = 0
+        dig_check = False
+        try:
+            int(vat)
```
Comment (jco-odoo) [inline addons/base_vat/models/res_partner.py]: isdigit?

File: addons/l10n_pe/__manifest__.py (hunk 1 above)
Comment (nhomar) [reply addons/l10n_pe/__manifest__.py]: move this dependency to edi!

File: addons/l10n_pe/models/res_company.py (hunk 2 above)
Comment (nhomar) [reply addons/l10n_pe/models/res_company.py]: Complimented with the _load technique used on cl/ar

**Comment by nhomar** [comment]
@jco-odoo superseeded by: https://github.com/odoo/odoo/pull/38764/files

File: addons/l10n_pe/__manifest__.py (hunk 1 above)
Comment (nhomar) [reply addons/l10n_pe/__manifest__.py]: Done

File: addons/l10n_pe/models/res_company.py (hunk 2 above)
Comment (nhomar) [reply addons/l10n_pe/models/res_company.py]: done

File: addons/l10n_pe/models/account_chart_template.py (hunk 3 above)
Comment (nhomar) [reply addons/l10n_pe/models/account_chart_template.py]: Removed this method, neither the account was necessary and the field will be set in the edi!

File: addons/base_vat/models/res_partner.py (hunk 4 above)
Comment (nhomar) [reply addons/base_vat/models/res_partner.py]: fully refactor DONE

**Pull request closed without merging**